epix/run_epix.py

```diff
@@ -0,0 +1,202 @@
+                                            epix.awkward_to_flat_numpy(result.y),
+                                            epix.awkward_to_flat_numpy(result.z)
+                                            )
+
+    result['e_field'] = epix.reshape_awkward(efields, ak.num(result))
+
+    # Sort in time and set first cluster to t=0, then chop all delayed
+    # events which are too far away from the rest.
+    # (This is a requirement of WFSim)
+    result = result[ak.argsort(result['t'])]
+
+    if args['debug']:
+        print('Generating photons and electrons for events')
+    # Generate quanta:
+    photons, electrons = epix.quanta_from_NEST(epix.awkward_to_flat_numpy(result['ed']),
+                                               epix.awkward_to_flat_numpy(result['nestid']),
+                                               epix.awkward_to_flat_numpy(result['e_field']),
+                                               epix.awkward_to_flat_numpy(result['A']),
+                                               epix.awkward_to_flat_numpy(result['Z']),
+                                               epix.awkward_to_flat_numpy(result['create_S2']),
+                                               density=epix.awkward_to_flat_numpy(result['xe_density']))
+    result['photons'] = epix.reshape_awkward(photons, ak.num(result['ed']))
+    result['electrons'] = epix.reshape_awkward(electrons, ak.num(result['ed']))
+    if args['debug']:
+        _ = monitor_time(tnow, 'get quanta.')
+
+    # Separate event in time
+    number_of_events = len(result["t"])
+    if args['source_rate'] == -1:
+        # Only needed for a clean separation:
+        result['t'] = result['t'] - result['t'][:, 0]
+        result = result[result['t'] <= args['max_delay']]
+
+        dt = epix.times_for_clean_separation(number_of_events, args['max_delay'])
+        if args['debug']:
+            print('Clean event separation')
+    elif args['source_rate'] == 0:
+        # In case no delay should be applied we just add zeros
+        dt = np.zeros(number_of_events)
+    else:
+        # Rate offset computed based on the specified rate and job_id.
+        # Assumes all jobs were started with the same number of events.
+        offset = (args['job_id']*n_simulated_events)/args['source_rate']
+        dt = epix.times_from_fixed_rate(args['source_rate'],
+                                        number_of_events,
+                                        n_simulated_events,
+                                        offset
+                                        )
+        if args['debug']:
+            print(f"Fixed event rate of {args['source_rate']} Hz")
+
+    result['t'] = result['t'][:, :] + dt
+
+    # Reshape instructions:
+    instructions = epix.awkward_to_wfsim_row_style(result)
+    if not args['source_rate'] == 0:
+        # Only sort by time if source rates are applied.
+        instructions = np.sort(instructions, order='time')
```

Comment:
Why do we have to sort here again? See line 106

Reply:
Probably the one in line 106 needs to go.

Reply:
I have to think maybe yes. You need the second sort since you may add random time numbers. But the first one may not be needed anymore.

Reply:
@WenzDaniel We still need to delete the first sorting, don't we?

Reply:
No, I do not think so. The first sorting is within the individual events, while the second sorting is done "globally". Hence, the first `ak.argsort` sorting is not strictly needed, but it also does not hurt either since it is fast. It just might make developers lives easier as they can assume things are sorted.